docs/tutorials/sparse/csr.md

```diff
@@ -0,0 +1,258 @@
+```
+
+Note that in MXNet, the column indices for a given row are always sorted in ascending order,
+and duplicated column entries for the same row are not allowed.
+
+## Array Creation
+
+There are a few different ways to create a `CSRNDArray`.
+
+* We can create a CSRNDArray with data, indices and indptr by using the `csr_matrix` function:
+
+```python
+import mxnet as mx
+import numpy as np
+# create a CSRNDArray with python lists
+shape = (3, 4)
+data_list = [7, 8, 9]
+indptr_list = [0, 2, 2, 3]
+indices_list = [0, 2, 1]
+a = mx.nd.sparse.csr_matrix(data_list, indptr_list, indices_list, shape)
+# create a CSRNDArray with numpy arrays
+data_np = np.array([7, 8, 9])
+indptr_np = np.array([0, 2, 2, 3])
+indices_np = np.array([0, 2, 1])
+b = mx.nd.sparse.csr_matrix(data_np, indptr_np, indices_np, shape)
+{'a':a, 'b':b}
+```
+
+* We can also create an MXNet CSRNDArray from a `scipy.sparse.csr.csr_matrix` object by using the `array` function:
+
+```python
+try:
+    import scipy.sparse as spsp
+    # generate a csr matrix in scipy
+    c = spsp.csr.csr_matrix((data_np, indices_np, indptr_np), shape=shape)
+    # create a CSRNDArray from a scipy csr object
+    d = mx.nd.sparse.array(c)
+    {'d':d}
+except ImportError:
+    print("scipy package is required")
+```
+
+We can specify the element data type with the option `dtype`, which accepts a numpy
+type. By default, `float32` is used.
+
+```python
+# float32 is used in default
+e = mx.nd.sparse.array(a)
+# create a 16-bit float array
+f = mx.nd.array(a, dtype=np.float16)
+(e.dtype, f.dtype)
+```
+
+## Inspecting Arrays
+
+* We can inspect the contents of an `CSRNDArray` by filling
+its contents into a dense `numpy.ndarray` using the `asnumpy` function.
+
+```python
+a.asnumpy()
+```
+
+* We can also inspect the internal storage of a CSRNDArray by accessing attributes such as `indptr`, `indices` and `data`:
+
+```python
+# access data array
+data = a.data
+# access indices array
+indices = a.indices
+# access indptr array
+indptr = a.indptr
+{'a.stype': a.stype, 'data':data, 'indices':indices, 'indptr':indptr}
+```
+
+## Storage Type Conversion
+
+* We can convert an NDArray to a CSRNDArray and vice versa by using the ``tostype`` function:
+
+```python
+# create a dense NDArray
+ones = mx.nd.ones((2,2))
+# cast the storage type from default to csr
+csr = ones.tostype('csr')
+# cast the storage type from csr to default
+dense = csr.tostype('default')
+{'csr':csr, 'dense':dense}
+```
+
+* We can also convert the storage type by using the ``cast_storage`` operator:
+
+```python
+# create a dense NDArray
+ones = mx.nd.ones((2,2))
+# cast the storage type to csr
+csr = mx.nd.sparse.cast_storage(ones, 'csr')
+# cast the storage type to default
+dense = mx.nd.sparse.cast_storage(csr, 'default')
+{'csr':csr, 'dense':dense}
+```
+
+## Copies
+
+* The `copy` method makes a deep copy of the array and its data, and `copyto` method or the slice
+operator `[]` to copy to an existing array.
```

Comment:
Did you mean: The copy method makes a deep copy of the array and its data, whereas the copyto method or the slice operator [] makes a reference to an existing array.

Reply:
Both are deep copy. I will clarify on this.